React/CxxBridge/RCTCxxBridge.mm

```diff
@@ -347,7 +355,21 @@ - (void)start
       executorFactory = [cxxDelegate jsExecutorFactoryForBridge:self];
     }
     if (!executorFactory) {
-      executorFactory = std::make_shared<JSCExecutorFactory>(nullptr);
+      auto installBindings =
```

Comment:
Should this block be inside the #else (!RCT_USE_HERMES) ?

Reply:
Both the `HermesExecutorFactory` and the `JSCExecutorFactory` need this block, which is why I moved it out of `JSCExecutorFactory.mm` into here.

Reply:
The previous implementation just did `executorFactory = std::make_shared<JSCExecutorFactory>(nullptr);`

I tested it locally: the same log events happen, i.e. _RCTLogJavaScriptInternal gets called for all logs, but they happen in a different callstack frame:

With `executorFactory = std::make_shared<JSCExecutorFactory>(installBindings);` the callstack for logs are:
```
#0	_RCTLogJavaScriptInternal
#1	-[RCTCxxBridge start]::$_0::operator()(facebook::jsi::Runtime&) const::'lambda'(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int)::operator()(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) const 
#2	decltype(std::__1::forward<-[RCTCxxBridge start]::$_0::operator()(facebook::jsi::Runtime&) const::'lambda'(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int)&>(fp)(std::__1::forward<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>(fp0), std::__1::forward<unsigned int>(fp0))) std::__1::__invoke<-[RCTCxxBridge start]::$_0::operator()(facebook::jsi::Runtime&) const::'lambda'(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int)&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int>(-[RCTCxxBridge start]::$_0::operator()(facebook::jsi::Runtime&) const::'lambda'(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int)&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int&&) at /Applications/Xcode_11.5.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:4425
...
``` 
With `executorFactory = std::make_shared<JSCExecutorFactory>(nullptr);` the callstack for logs are:
```
#0 _RCTLogJavaScriptInternal 
#1 facebook::react::JSCExecutorFactory::createJSExecutor(std::__1::shared_ptr<facebook::react::ExecutorDelegate>, std::__1::shared_ptr<facebook::react::MessageQueueThread>)::$_0::operator()(facebook::jsi::Runtime&) const::'lambda'(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int)::operator()(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) const 
#2 decltype(std::__1::forward<facebook::react::JSCExecutorFactory::createJSExecutor(std::__1::shared_ptr<facebook::react::ExecutorDelegate>, std::__1::shared_ptr<facebook::react::MessageQueueThread>)::$_0::operator()(facebook::jsi::Runtime&) const::'lambda'(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int)&>(fp)(std::__1::forward<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&>(fp0), std::__1::forward<unsigned int>(fp0))) std::__1::__invoke<facebook::react::JSCExecutorFactory::createJSExecutor(std::__1::shared_ptr<facebook::react::ExecutorDelegate>, std::__1::shared_ptr<facebook::react::MessageQueueThread>)::$_0::operator()(facebook::jsi::Runtime&) const::'lambda'(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int)&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int>(facebook::react::JSCExecutorFactory::createJSExecutor(std::__1::shared_ptr<facebook::react::ExecutorDelegate>, std::__1::shared_ptr<facebook::react::MessageQueueThread>)::$_0::operator()(facebook::jsi::Runtime&) const::'lambda'(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int)&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int&&) 
...
```
I don't think it matters.   I was just worried that we might be getting some new logging events, or duplicated logging events, but we're not: we're getting identical logging events, and in the same order, just from a slightly different callstack.  So I'm ok with it.

Reply:
Thanks for being vigilant!